postgres_base_block.py

```diff
@@ -60,6 +61,13 @@ def stop(self):
         self.disconnect()
         super().stop()
 
+    def process_signals(self, signals):
+        self.execute_with_lock(
+            self._locked_process_signals, 10, signals=signals)
```

Comment:
We may have to tweak the number of locks this allows, max locks was achieved pretty quickly on my local test.